src/server/game/Entities/Player/Player.cpp

```diff
@@ -15103,6 +15109,16 @@ void Player::PrepareGossipMenu(WorldObject* source, uint32 menuId /*= 0*/, bool
                     }
                     break;
                 case GOSSIP_OPTION_GOSSIP:
+                    if (creature->isVendorWithIconSpeak())
+                    {
+                        VendorItemData const* vendorItems = itr->second.ActionMenuID ? sObjectMgr->GetNpcVendorItemList(itr->second.ActionMenuID) : creature->GetVendorItems();
+                        if (!vendorItems || vendorItems->Empty())
+                        {
+                            sLog->outErrorDb("Creature %u (Entry: %u) have UNIT_NPC_FLAG_VENDOR but have empty trading item list.", creature->GetGUIDLow(), creature->GetEntry());
+                            canTalk = false;
+                        }
+                    }
+                    break;
```

Comment:
the GOSSIP_OPTION_GOSSIP is only for vendor? are you sure that you are not breaking the other GOSSIP NPCs?

Reply:
No this flag isn't only for vendors, but for anyone with option gossip. I'm not breaking the others for 2 reasons:
1. This case was empty before;
2. I only check for non player characters with flag gossip and icon Speak.